event-protocol/schemas/defs.json

```diff
@@ -38,6 +38,49 @@
         "type": "string"
       },
       "location": { "$ref": "defs.json#/location" }
-    }
+    },
+    "required": [
+      "name",
+      "location"
+    ],
+    "additionalProperties": false
+  },
+  "ast-location": {
+    "type": "object",
+    "properties": {
+      "line": {
+        "type": "integer",
+        "minimum": 1
+      },
+      "column": {
+        "type": "integer",
+        "minimum": 1
+      }
+    },
+    "required": [
+      "line",
+      "column"
+    ],
+    "additionalProperties": false
+  },
+  "ast-tag": {
```

Comment:
The JSON representation of an AST Tag and a Pickle Tag are slightly different. The Java and .NET implementations of Gherkin even have two different classes (the generated JSON depends on the fields in those classes):

* [Tag](https://github.com/cucumber/gherkin/blob/master/java/src/main/java/gherkin/ast/Tag.java) which inherits fields from [Node](https://github.com/cucumber/gherkin/blob/master/java/src/main/java/gherkin/ast/Node.java)
* [PickleTag](https://github.com/cucumber/gherkin/blob/master/java/src/main/java/gherkin/pickles/PickleTag.java)

This `type` attribute on AST nodes is useful in some corner cases, where a collection can have different types. For example, the `Feature.children` collection can have 3 different types of elements - `Background`, `Scenario` and `ScenarioOutline`. The `type` attribute makes it easy to know what type it is.

For pickles we don't have this problem, because it's a flatter, simpler structure where we don't have this polymorphism.

Since `Tag` != `PickleTag` I thought it made more sense to have distinct JSON schemas for them, even if they look similar. It's less confusing, and also allows us to change one of them without having to change the other.